streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -0,0 +1,1123 @@
+        //        * the second client will have four pending transactions
+        // 5.  restart the first client with eos-beta enabled and wait until rebalance stabilizes
+        //       - the rebalance should result in a commit of all tasks
+        // 6.  write 5 record per input topic partition
+        //       - stop case:
+        //         * verify that the result was committed
+        //       - crash case:
+        //         * fail the second (i.e., eos-alpha) client during commit
+        //         * the eos-beta client should not pickup the pending offsets
+        //         * verify uncommitted and committed result
+        // 7.  only for crash case:
+        //     7a. restart the second client in eos-alpha mode and wait until rebalance stabilizes
+        //     7b. write 10 records per input topic partition
+        //         * fail the first (i.e., eos-beta) client during commit
+        //         * the eos-alpha client should not pickup the pending offsets
+        //         * verify uncommitted and committed result
+        //     7c. restart the first client in eos-beta mode and wait until rebalance stabilizes
+        // 8.  write 5 records per input topic partition to get pending transactions (verified via "read_uncommitted" mode)
+        //      - 2 transaction are base on a task producer; one transaction is based on a thread producer
+        //      - we will get 4 pending writes for the crash case as we crash processing the 5th record
+        // 9.  stop/crash the second client and wait until rebalance stabilizes:
+        //      - stop only:
+        //        * verify that the stopped client did commit its pending transaction during shutdown
+        //        * the first client will still have one pending transaction
+        //      - crash case:
+        //        * the pending transactions of the crashed client got aborted
+        //        * the first client will have one pending transactions
+        // 10. restart the second client with eos-beta enabled and wait until rebalance stabilizes
+        //       - the rebalance should result in a commit of all tasks
+        // 11. write 5 record per input topic partition and verify that the result was committed
+
+        final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> stateTransitions1 = new LinkedList<>();
+        KafkaStreams streams1Alpha = null;
+        KafkaStreams streams1Beta = null;
+        KafkaStreams streams1BetaTwo = null;
+
+        final List<KeyValue<KafkaStreams.State, KafkaStreams.State>> stateTransitions2 = new LinkedList<>();
+        KafkaStreams streams2Alpha = null;
+        KafkaStreams streams2AlphaTwo = null;
+        KafkaStreams streams2Beta = null;
+//        streams2Beta = getKafkaStreams("appDir2", StreamsConfig.EXACTLY_ONCE_BETA);
+//        streams2Beta.setStateListener((newState, oldState) -> stateTransitions2.add(KeyValue.pair(oldState, newState)));
+
+        try {
```

Comment:
We use 6 clients now, to do some error injection in "mixed mode". To avoid JXM warnings, we cannot create all clients at the same time and thus cannot use try-with-resources...